hamilton/plugins/huggingface_extensions.py

```diff
@@ -0,0 +1,206 @@
+import dataclasses
+from os import PathLike
+from typing import Any, BinaryIO, Collection, Dict, Mapping, Optional, Sequence, Tuple, Type, Union
+
+try:
+    from datasets import (
+        Dataset,
+        DatasetDict,
+        DownloadConfig,
+        DownloadMode,
+        Features,
+        IterableDataset,
+        IterableDatasetDict,
+        VerificationMode,
+        Version,
+        load_dataset,
+    )
+    from datasets.formatting.formatting import LazyBatch
+except ImportError:
+    raise NotImplementedError("huggingface datasets library is not installed.")
+
+try:
+    import lancedb
+    from lancedb import table  # noqa: F401
+except ImportError:
+    lancedb = None
+
+from hamilton import registry
+from hamilton.io import utils
+from hamilton.io.data_adapters import DataLoader, DataSaver
+
+COLUMN_FRIENDLY_DF_TYPE = False
+
+HF_types = (DatasetDict, Dataset, IterableDatasetDict, IterableDataset)
+
+
+@dataclasses.dataclass
+class HuggingFaceDSLoader(DataLoader):
+    path: str
+    dataset_name: Optional[str] = None  # this can't be `name` because it clashes with `.name()`
+    data_dir: Optional[str] = None
+    data_files: Optional[Union[str, Sequence[str], Mapping[str, Union[str, Sequence[str]]]]] = None
+    split: Optional[str] = None
+    cache_dir: Optional[str] = None
+    features: Optional[Features] = None
+    download_config: Optional[DownloadConfig] = None
+    download_mode: Optional[Union[DownloadMode, str]] = None
+    verification_mode: Optional[Union[VerificationMode, str]] = None
+    ignore_verifications = "deprecated"
+    keep_in_memory: Optional[bool] = None
+    save_infos: bool = False
+    revision: Optional[Union[str, Version]] = None
+    token: Optional[Union[bool, str]] = None
+    use_auth_token = "deprecated"
+    task = "deprecated"
+    streaming: bool = False
+    num_proc: Optional[int] = None
+    storage_options: Optional[Dict] = None
+    config_kwargs: Optional[Dict] = None
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return list(HF_types)
+
+    def _get_loading_kwargs(self) -> dict:
```

Comment:
Might want to look at this approach: https://github.com/DAGWorks-Inc/hamilton/blob/818f2fca8b800b1a2b66fea31fdd73d5534beed8/hamilton/plugins/dlt_extensions.py#L96

I think `fields_to_skip` is a clear approach. Could document further why to skip in the docstring

Reply:
I tried it -- and it's more lines of code in the other cases. So not going to use it here.